src/Database/LSMTree/Internal/WriteBuffer.hs

```diff
@@ -116,11 +170,11 @@ lookups ::
      WriteBuffer
   -> V.Vector SerialisedKey
   -> V.Vector (Maybe (Entry SerialisedValue SerialisedBlob))
-lookups (WB !m) !ks = V.mapStrict (`Map.lookup` m) ks
+lookups (WB m _) !ks = V.mapStrict (`Map.lookup` m) ks
 
 -- | TODO: update once blob references are implemented
 lookup :: WriteBuffer -> SerialisedKey -> Maybe (Entry SerialisedValue (BlobRef run))
-lookup (WB !m) !k = case Map.lookup k m of
```

Comment:
Do we need to keep these bang patterns? Even if it is only to be extra sure about strictness?